### ISSUE TITLE
Improper IDE0004 Cast is Redundant Suggestion for Delegates Conversion

### ISSUE THREAD

**Issue by fitdev** [dotnet/roslyn#62072]
**Version Used**: 
Latest: VS 2022 17.3 Preview 2
CS 11 Preview on DotNet 6

**Steps to Reproduce**:

```cs
#nullable enable

using System;
public class C<T> {
    public C() {}
    public C(bool foo, Func<T, bool> predicate) { Foo = foo; Predicate = predicate; }
    public bool Foo {get; set; }
    public Func<T, bool>? Predicate {get; set;}
    public static implicit operator C<T>(Func<T, bool> predicate) => new(false, predicate);
}

public class UseC {
 public static void Main() {
     C<int> works = (Func<int,bool>)((int i) => i > 10); //<= Wrong Suggestion Here: Cast is in fact NOT redundant
     C<int> doesnt = ((int i) => i > 10); //<= Without a cast, this is a compile error!
 }
}
```

([Sharplab Repro](https://sharplab.io/#v2:C4LghgzgtgPgxAOwK4BsVgEYoKYAJsKY4CwAUGQAIBMAjJQMy7W4DCAPACoB8uA3mbkFNGLABQBKPgF8BQiiNEYA9kpS4AZioA0TAKycdy1TwAOAJ2wATAJYBjMMGyTeuAGIrcAXg0qA3LgAFCxt7Ry9ccys7B2x/GVIhYVwjNXclPgBzbGB/CGy42UF5PQNklRQuAH5A4Oiw3iyc3Dyc+MTiihoANlxrKBMUO2tgXCUTbDMHJTNWTi5RCn0OQ3LTWtCnLx4EbAB3UXUwFDydSJCY8V8yeIYmKlwAVTyWPgFqe/4ExPZrBGAeXbTADWEHCC30v2AWhSXHEolEkN6kk8PGsuB4NAADJdcAB6XFsbwAdTMSgQGVwAGUkBkshBgNYybgABITbAgViQEbWUG/DRgWwjAByAHkOLhgkgEJYwH9CoIfn8eJYlNgIH8wQiNdZkaj0bgsTj8YTcEThgALJRIEZgXD2ek6YDmnm9UG22xKfrWHD4MykswAQgE8SkQA===))

**Expected Behavior**:
IDE should _not_ suggest that the cast is redundant! Well, actually even better, though I realize it's a cslang issue, but even better would be that CS would automatically take care of that and allow implicit conversion.

**Actual Behavior**:
IDE suggests the cast as redundant.

**Comment by CyrusNajmabadi** [comment]
@cston ?

**Comment by js6pak** [comment]
Looks like this got fixed with #72135
